### PR TITLE
Return no effect when *Solid* value is returned by the device

### PR DESCRIPTION
**Pull request by Taosmi**
Canvas device returns the value '*Solid*' as current effect selected but there is really no effect selected.

### CODE REVIEW

**Comment by Oro** [comment]
Awesome, thank you! I completely missed that.  

Reading the documentation, `*Dynamic*` and `*Static*` are also reserved effect names, could you please also add them to return None? Maybe with something like the following and returning None if any of them? I'll release soon afterwards (as version 0.1.0 seeing as this is a breaking change)

```python3
RESERVED_EFFECT_NAMES = ["*Solid*", "*Static*", "*Dynamic*"] # These effect names have special meaning and are not directly setable and should not be returned as effect name for Consumers relying on the effect returned to be included in the list of all effects
```

**Comment by Taosmi** [comment]
Great. I just added the suggested change and effects and effectSelect on the Info dict to make it easy to use.

File: setup.py [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 setup(
     name='pynanoleaf',
-    version='0.0.7',
```
Comment (Oro) [inline setup.py]: No need to up the version number here, I'll do that after this PR! Please leave that out.

File: pynanoleaf/pynanoleaf.py [hunk 2]
```diff
@@ -40,6 +52,10 @@ class _Info(dict):
     firmwareVersion = _dictitem_property('firmwareVersion')
     model = _dictitem_property('model')
     name = _dictitem_property('name')
+    manufacturer = _dictitem_property('manufacturer')
```
Comment (Oro) [inline pynanoleaf/pynanoleaf.py]: That is an awesome change, could you please do this in a separate PR? Makes it easier to review :)

File: pynanoleaf/pynanoleaf.py [hunk 3]
```diff
@@ -9,6 +9,8 @@ def _deepgetter(key):
     def get(data):
         for k in key:
             data = data[k]
+        if ('.'.join(key) == 'effects.select'):
```
Comment (Oro) [inline pynanoleaf/pynanoleaf.py]: That is an awesome change, could you please do this in a separate PR? Makes it easier to review :)

File: pynanoleaf/pynanoleaf.py [hunk 4]
```diff
@@ -32,6 +34,16 @@ def _dictitem_property(keystr):
         fset=_deepsetter(key)
     )
 
+def _parseEffect(effect):
+    """These effect names have special meaning and are not directly setable
+    and should not be returned as effect name for Consumers."""
+    RESERVED_EFFECT_NAMES = ["*Solid*", "*Static*", "*Dynamic*"]
```
Comment (Oro) [inline pynanoleaf/pynanoleaf.py]: I was thinking of putting this at the top (like a global variable?), would that be better, what do you think?

File: pynanoleaf/pynanoleaf.py [hunk 5]
```diff
@@ -32,6 +34,16 @@ def _dictitem_property(keystr):
...
+    try:
```
Comment (Oro) [inline pynanoleaf/pynanoleaf.py]: Any particular reason a try/except is used here? If not, could you please change it to something like the following:
```python
if effect in RESERVED_EFFECT_NAMES
  return None
else
  return effect
```

**Review by Oro — CHANGES_REQUESTED** [review]
Thanks again for the PR and the changes, a couple of notes though.

File: pynanoleaf/pynanoleaf.py (hunk 3 above)
Comment (keis) [reply pynanoleaf/pynanoleaf.py]: Hey, I'll chime in here too :)

I think it's better to keep this utility function free of any logic regarding specific fields. Instead write a regular property for this special case.

```
@property
def effectSelect(self):
    return _parseEffect(self['effects']['select'])
```

File: pynanoleaf/pynanoleaf.py (hunk 4 above)
Comment (keis) [reply pynanoleaf/pynanoleaf.py]: it's also short enough to write inline imho

File: setup.py (hunk 1 above)
Comment (Taosmi) [reply setup.py]: ok

File: pynanoleaf/pynanoleaf.py (hunk 2 above)
Comment (Taosmi) [reply pynanoleaf/pynanoleaf.py]: ok

File: pynanoleaf/pynanoleaf.py (hunk 4 above)
Comment (Taosmi) [reply pynanoleaf/pynanoleaf.py]: ok

File: pynanoleaf/pynanoleaf.py (hunk 3 above)
Comment (Taosmi) [reply pynanoleaf/pynanoleaf.py]: Ok

File: pynanoleaf/pynanoleaf.py (hunk 5 above)
Comment (Oro) [reply pynanoleaf/pynanoleaf.py]: @Taosmi this is still open, probably an oversight. Could you please double-check?

Comment (Taosmi) [reply pynanoleaf/pynanoleaf.py]: ok

**Comment by keis** [comment]
Nice work @Taosmi
